src/hotspot/share/runtime/thread.cpp

```diff
@@ -1670,9 +1670,13 @@ class InstallAsyncExceptionHandshake : public HandshakeClosure {
 public:
   InstallAsyncExceptionHandshake(AsyncExceptionHandshake* aeh) :
     HandshakeClosure("InstallAsyncException"), _aeh(aeh) {}
+  ~InstallAsyncExceptionHandshake() {
+    delete _aeh;
```

Comment:
A comment would have been good. I'm assuming:

// Can only be non-null if this handshake was never actually executed.

Reply:
Since we clear `_aeh` on L1679 below when `do_thread()` is executed, `_aeh` can
only be non-NULL when `~InstallAsyncExceptionHandshake()` runs if `do_thread()`
was never executed.

I can add that comment above L1674 in a separate bug fix if you wish.
@robehn - do you concur that a comment here would be good?

Reply:
It was the fact we null it and delete it that initially caused a "What the???" reaction. :) So I think I comment would be good but can wait.

Reply:
Yes add a comment.

Reply:
Filed:

JDK-8286088 add comment to InstallAsyncExceptionHandshake destructor
https://bugs.openjdk.java.net/browse/JDK-8286088